core-feature-pack/src/main/resources/content/bin/common.ps1

```diff
@@ -0,0 +1,256 @@
+$global:MODULE_OPTS = @()
+
+# A collection of functions that are used by the other scripts
+
+Function Set-Env {
+  $key = $args[0]
+  $value = $args[1]
+  Set-Content -Path env:$key -Value $value
+}
+
+Function Get-Env {
+  $key = $args[0]
+  if( Test-Path env:$key ) {
+    return (Get-ChildItem env:$key).Value
+  }
+  return $args[1]
+}
+
+Function Get-String {
+  $value = ''
+  foreach($k in $args) {
+    $value += $k
+  }
+  return $value
+}
+
+Function String-To-Array($value) {
+  $res = @('')
+  if (-Not(Test-Path $value)){
+  	return $res
+  }
+  $tmpArr = $value.split()
+  
+  foreach ($str in $tmpArr) {
+    if ($str -ne '') {
+	  $res += $str
+	}
+  }
+  return $res
+}
+
+Function Display-Environment { 
+
+# Display our environment
+echo "================================================================================="
+echo ""
+echo "  WildFly Bootstrap Environment"
+echo ""
+echo "  JBOSS_HOME: $JBOSS_HOME"
+echo ""
+echo "  JAVA: $JAVA"
+echo ""
+echo "  MODULE_OPTS: $MODULE_OPTS"
+echo ""
+echo "  JAVA_OPTS: $JAVA_OPTS"
+echo ""
+echo "================================================================================="
+echo ""
+
+}
+
+Function Get-Java-Arguments {
+Param(
+   [Parameter(Mandatory=$true)]
+   [string]$entryModule,
+   [string]$logFileProperties = "$JBOSS_CONFIG_DIR/logging.properties",
+   [string]$logFile = "$JBOSS_LOG_DIR/server.log",
+   [string[]]$serverOpts
+   
+
+) #end param
+
+  $PROG_ARGS = @()
+  if ($JAVA_OPTS -ne $null){
+  	$PROG_ARGS += $JAVA_OPTS
+  }  
+  $PROG_ARGS += "-Dorg.jboss.boot.log.file=$logFile"
+  $PROG_ARGS += "-Dlogging.configuration=file:$logFileProperties"
+  $PROG_ARGS += "-Djboss.home.dir=$JBOSS_HOME"
+  $PROG_ARGS += "-jar"
+  $PROG_ARGS += "$JBOSS_HOME\jboss-modules.jar"
+  if ($MODULE_OPTS -ne $null){
+  	$PROG_ARGS += $MODULE_OPTS
+  }
+  $PROG_ARGS += "-mp"
+  $PROG_ARGS += "$JBOSS_MODULEPATH"
+  $PROG_ARGS += $entryModule
+  if ($serverOpts -ne $null){
+  	$PROG_ARGS += $serverOpts
+  }
+  return $PROG_ARGS
+}
+
+Function Process-Script-Parameters {
+Param(
+   [Parameter(Mandatory=$false)]
+   [string[]]$Params   
+
+) #end param
+    $res = @()
+	for($i=0; $i -lt $Params.Count; $i++){
+		$arg = $Params[$i]
+		if ($arg -eq '--debug'){
+			$global:DEBUG_MODE=$true
+			if ($args[$i+1] -match '\d+'){ #port number can only follow --debug
+				$global:DEBUG_PORT = $Params[$i+1]
+				$i++
+				continue
+			}
+		}elseif ($arg -contains '-Djava.security.manager'){
+			echo "ERROR: The use of -Djava.security.manager has been removed. Please use the -secmgr command line argument or SECMGR=true environment variable."
+			exit
+		}elseif ($arg -eq '-secmgr'){
+			$global:SECMGR = $true
+		}elseif ($arg -eq '--background'){
+			$global:RUN_IN_BACKGROUND = $true
+		}else{
+			$res+=$arg
+		}
+	}
+	return $res
+}
+
+Function Start-WildFly-Process {
+ Param(
+   [Parameter(Mandatory=$true)]
+   [string[]] $programArguments,
+   [boolean] $runInBackground = $false   
```

Comment:
Does this mean `false` is a default or does it currently always set `runInBackground` to `false`?